decidim-core/spec/tasks/upgrade/clean_action_logs_spec.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "rake decidim:upgrade:clean:action_logs", type: :task do
+  context "when executing task" do
+    it "does not throw exceptions keys" do
+      expect { task.execute }.not_to raise_exception
+    end
+
+    context "when there are no errors" do
+      let!(:action_logs) { create_list(:action_log, 8, created_at: 2.days.ago) }
+
+      it "avoid removing entries" do
+        expect { task.execute }.not_to change(Decidim::ActionLog, :count)
+      end
+    end
+
+    context "when there are errors" do
+      let!(:action_logs) { create_list(:action_log, 8, created_at: 2.days.ago) }
+      let!(:invalid_entries) { action_logs.collect(&:id).sample(4) }
+
+      context "when errors in Component" do
+        it "removes entries" do
+          Decidim::ActionLog.where(id: invalid_entries).update_all(resource_type: "Decidim::Component", resource_id: Time.now.to_i) # rubocop:disable Rails/SkipsModelValidations
```

Comment:
@andreslucena Running this spec in parallel creates some issues here. Dunno why, but i needed to push the component id a little bit further, so that we know for sure it does not exists.